website/docs/error-reference.md

```diff
@@ -1803,6 +1803,63 @@ def foo; [0]; end
 For more information, see
 [Arrays, Hashes, and Generics in the Standard Library](stdlib-generics.md).
 
+## 5027
```

Comment:
```suggestion
## 5027

> This error is opt-in, behind the `--check-out-of-order-constant-references`
> flag.
>
> Sorbet does not check this by default because certain codebases make clever
> usage of Ruby's `autoload` mechanism to allow all constants to be referenced
> before their definitions.
```